EIP-008-plugin-call-ergonomics.md

```diff
@@ -0,0 +1,21 @@
+#  Name of proposal
+
+Plugin Call Ergonomics
+
+## Purpose
+
+Evaluate possible improvements to the ergonomics of calling plugin functions.
+
+Some notes for discussion:
+
+1) Instead of the plugin function selection being stringly-typed (eww), make it typed using a mechanism appropriate to each language (e.g. enums)
```

Comment:
Interesting – IIUC, this would require a bit of dynamic type generation in languages that support it? (Since we don't know what functions are exported from a Wasm file until we load it at run time.)

Reply:
Are y'all planning on supporting packaging plugins into languages's respective package managers? I hope so :)
Because that would make it easier to support including plugin API metadata that could be leveraged by the plugin calling mechanism.

Reply:
AFAIK, we don't currently have plans to do this – which is not to say it's not on the table! @bhelx might know more about this.